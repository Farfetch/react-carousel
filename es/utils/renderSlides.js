import { Children } from 'react';
import { cloneSlide } from "./";

var renderSlides = function renderSlides(children, itemsToShow, style) {
  return Children.map(children, function (child, index) {
    return cloneSlide(child, index, false, null, style);
  });
};

export default renderSlides;