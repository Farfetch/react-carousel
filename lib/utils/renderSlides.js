"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = require("react");

var _ = require("./");

var renderSlides = function renderSlides(children, itemsToShow, style) {
  return _react.Children.map(children, function (child, index) {
    return (0, _.cloneSlide)(child, index, false, null, style);
  });
};

var _default = renderSlides;
exports["default"] = _default;
module.exports = exports.default;