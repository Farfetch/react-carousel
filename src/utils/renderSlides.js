import { Children } from 'react';
import { cloneSlide } from '#utils';

const renderSlides = (children, itemsToShow, style) => {
    return Children.map(children, (child, index) =>
        cloneSlide(child, index, false, null, style)
    );
};

export default renderSlides;
