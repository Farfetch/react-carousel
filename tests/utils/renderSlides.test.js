import React from 'react';
import renderSlides from '../../src/utils/renderSlides';

describe('renderSlides()', () => {
    test('should return slides array', () => {
        const children = [
            <div key="0">First slide</div>,
            <div key="1">Second slide</div>,
            <div key="2">Third slide</div>,
            <div key="3">Fourth slide</div>,
        ];

        const itemsToShow = 2;
        const style = { width: `${(100 / itemsToShow).toFixed(6)}%` };

        const slides = renderSlides(children, itemsToShow, style);

        expect(slides).toMatchSnapshot();
        expect(slides.length).toBe(children.length);
    });
});
