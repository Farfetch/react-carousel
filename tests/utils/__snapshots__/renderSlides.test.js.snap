// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSlides() should return slides array 1`] = `
Array [
  <div
    className="slide"
    style={
      Object {
        "width": "50.000000%",
      }
    }
  >
    First slide
  </div>,
  <div
    className="slide"
    style={
      Object {
        "width": "50.000000%",
      }
    }
  >
    Second slide
  </div>,
  <div
    className="slide"
    style={
      Object {
        "width": "50.000000%",
      }
    }
  >
    Third slide
  </div>,
  <div
    className="slide"
    style={
      Object {
        "width": "50.000000%",
      }
    }
  >
    Fourth slide
  </div>,
]
`;
